### PR TITLE
Add the axpby! function

### PR DESCRIPTION
**Pull request by amontoison**

### CODE REVIEW

File: src/blas/highlevel.jl [hunk 1]
```diff
@@ -47,8 +47,19 @@ end
 LinearAlgebra.BLAS.asum(x::ROCBLASArray) = asum(length(x), x, 1)
 
 function LinearAlgebra.axpy!(alpha::Number, x::ROCArray{T}, y::ROCArray{T}) where T<:ROCBLASFloat
```
Comment (jpsamaroo) [inline src/blas/highlevel.jl:49]:
```suggestion
function LinearAlgebra.axpy!(alpha::Number, x::ROCArray{T}, y::ROCArray{T}) where T<:Union{ROCBLASFloat,ROCBLASComplex}
```
Has a fallback in GPUArrays, but let's not hit that.

File: src/blas/highlevel.jl [hunk 2]
```diff
@@ -47,8 +47,19 @@ end
...
-    length(x)==length(y) || throw(DimensionMismatch(""))
-    axpy!(length(x), convert(T,alpha), x, 1, y, 1)
+    lx = length(x)
+    ly = length(y)
+    lx==ly || throw(DimensionMismatch(""))
+    axpy!(lx, convert(T,alpha), x, 1, y, 1)
+end
+
+function LinearAlgebra.axpby!(alpha::Number, x::ROCArray{T}, beta::Number, y::ROCArray{T}) where T<:ROCBLASFloat
```
Comment (jpsamaroo) [inline src/blas/highlevel.jl:56]:
```suggestion
function LinearAlgebra.axpby!(alpha::Number, x::ROCArray{T}, beta::Number, y::ROCArray{T}) where T<:Union{ROCBLASFloat,ROCBLASComplex}
```

**Review by jpsamaroo — APPROVED** [review]
:+1:

**Comment by pxl-th** [comment]
Superseded by #484.

**Pull request closed without merging**